### PR TITLE
fix: Support Impression and Promotion events

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
This PR adds support for bundling impressions and promotions into a single event if multiple are sent in a single SDK call.  For impressions, we support having products as an array as well.

 ## Testing Plan
 - [X] Was this tested locally? If not, explain why.
Local testing in an app and ran unit tests.

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5472

### CODE REVIEW

File: src/BrazeKit-dev.js [hunk 1]
```diff
@@ -361,21 +366,84 @@ var constructor = function () {
         }
     }
 
-    function logNonPurchaseCommerceEventWithProducts(event) {
-        var sanitizedProperties = getSanitizedCustomProperties(
-            event.EventAttributes
-        );
-        var productArray = [];
+    function logNonPurchaseCommerceEventWithProducts(mpEvent) {
+        var brazeEvent = {
```
Comment (alexs-mparticle) [inline src/BrazeKit-dev.js:370]: I wonder if we can clean this up a bit by just creating a simpler object, `eventAttributes`, and build it within the switch case and then wrap it in the braze Event object in the end.

For example:

```
const eventAttributes = {};

// Switch Case

eventAttributes.promotion = addPromotions(mpEvent)

// At the end

const brazeEvent = {
  EventName: mpEvent.EventName,
  EventAttributes: mergeObjects(
    brazeEvent.EventAttributes,
    sanitizedProperties
  )
}
```

File: test/tests.js [hunk 2]
```diff
@@ -1698,11 +1698,8 @@ USD,
         purchaseEventProperties.should.eql(expectedPurchaseEvent);
     });
 
-    it('should log promotion events via the logCustomEvent method', function() {
-        // This initialization is a hack in order to get sessionId to exist, which
-        // expandCommerceEvent depends on
-
-        mParticle.forwarder.process({
+    describe('promoimpression eventstion events', function() {
```
Comment (alexs-mparticle) [inline test/tests.js:1701]: Are these new kinds of events?

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

File: src/BrazeKit-dev.js (hunk 1 above)
Comment (rmi22186) [reply src/BrazeKit-dev.js:370]: I like this suggestion

File: src/BrazeKit-dev.js [hunk 3]
```diff
@@ -361,21 +366,84 @@ var constructor = function () {
...
+            switch (mpEvent.EventCategory) {
+                case CommerceEventTypes.PromotionClick:
+                case CommerceEventTypes.PromotionView:
+                    brazeEvent.EventAttributes.promotions = addPromotions(
+                        mpEvent,
+                        brazeEvent
+                    );
+                    break;
+                case CommerceEventTypes.ProductImpression:
+                    brazeEvent.EventAttributes.impressions = addImpressions(
```
[the remark this answers is not in the document]
Comment (rmi22186) [reply src/BrazeKit-dev.js:384]: ditto

File: test/tests.js (hunk 2 above)
Comment (rmi22186) [reply test/tests.js:1701]: 😢

**Review by alexs-mparticle — APPROVED** [review]
:shipit:

**Pull request merged**